.circleci/config.yml

```diff
@@ -319,52 +319,63 @@ jobs:
         enum: ["chef", "puppet", "salt", "ansible"]
     steps:
       - checkout
-      - run: |
-          DEPLOYMENT=<< parameters.deployment_type >>
-          mkdir ~/testresults
-          if ! scripts/changes-include-dir Dockerfile deployments/$DEPLOYMENT; then
-              echo "$DEPLOYMENT code has not changed, skipping tests!"
-              exit 0
-          fi
-          cd deployments/$DEPLOYMENT
-          make dev-image
-          case "$DEPLOYMENT" in
-          chef)
-              CHEF_DEV="docker run --rm \
-                  --workdir /chef-repo/cookbooks/signalfx_agent \
-                  signalfx-agent-chef-dev"
+      - run:
+          name: Run << parameters.deployment_type >> tests
+          command: |
+            DEPLOYMENT=<< parameters.deployment_type >>
+            mkdir ~/testresults
+            if ! scripts/changes-include-dir Dockerfile deployments/$DEPLOYMENT tests/deployments/$DEPLOYMENT; then
+                echo "$DEPLOYMENT code has not changed, skipping tests!"
+                touch ~/.skip
+                exit 0
+            fi
+            if [ ! -d tests/deployments/$DEPLOYMENT ]; then
+                # no pytest tests to execute for deployment
+                touch ~/.skip
+            fi
+            cd deployments/$DEPLOYMENT
+            make dev-image
+            case "$DEPLOYMENT" in
+            chef)
+                CHEF_DEV="docker run --rm \
```

Comment:
I would swap the ordering here and run the linting stuff before the integration test for Chef.

Reply:
Since if you have lint errors it will require repushing whereas integration test failures might be transient and can be retried.

Reply:
This will run the linting stuff first.  `run_pytest` will run at the end of the job if linting is successful.